code/script.js

```diff
@@ -231,95 +236,120 @@ const setSecret = () => {
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  generateBoard();
+  setSecret();
+  selectQuestion();
 }
 
+
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
-
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const value = questions.value; //retrieves the selected question value from the dropdown and store it in the currentQuestion object 
 
   currentQuestion = {
     category: category,
-    // value: value
+    value: value
   }
-}
+} 
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
   const { category, value } = currentQuestion
-
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
   if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
-  }
+      if (secret[category] === value) {
+        filterCharacters(true);
+      } else {
+        filterCharacters(false);
+      }
+    
+  } else if (category  === 'accessories' || category === 'other') {
+      if (secret[category].includes(value)) {
+        filterCharacters(true);
+      } else {
+        filterCharacters(false);
+      }
+  }   
```

Comment:
It's a little bit hard for me to understand this part, but it's probably because I wrote it in a different way. But both ways work so it doesn't really matter! But perhaps a little comment to explain this step a bit?